src/components/NewArticle/index.js

```diff
@@ -0,0 +1,311 @@
+    const newTags = tagArray.slice();
+
+    // change tag position
+    newTags.splice(currPos, 1);
+    newTags.splice(newPos, 0, tag);
+
+    // re-render
+    this.setState({ tags: newTags });
+  }
+
+  render() {
+    const {
+      title, description, body, category, categoryOptions, tags, tagOptions
+    } = this.state;
+
+    return (
+      <div className="site-content">
+        <form className="edit-article-container">
+          <h3>Add a new article</h3>
+          <div className="title-cell">
+            <label>
+              <span>Title</span>
+              <input
+                id="title"
+                name="title"
+                className="title-input"
+                onChange={this.handleChange}
+                value={title} placeholder="Title"
+                type="text"
+                autoComplete="off" />
+            </label>
+          </div>
+          <div className="description-cell">
+            <label>
+              <span>Description</span>
+              <input
+                id="description" name="description"
+                onChange={this.handleChange}
+                value={description}
+                placeholder="Add description here"
+                type="text"
+                autoComplete="off" />
+            </label>
+          </div>
+
+          <div className="editor-cell">
+            <TextEditor body={body} onEditorChange={this.handleEditorChange} />
+          </div>
+
+          <div className="categories-cell">
+            <CategoryDropdown
+              category={category}
+              categoryOptions={categoryOptions}
+              onCategoryChange={this.handleCategoryChange} />
+          </div>
+
+          <div className="upload-cell">
+            <span>Image Upload</span>
+            <div id="selected-image-container" />
+            <button type="button" id="image-browse-button">
+              <label>
+                <input type="file" name="pic" accept="image/*" />
+                Upload
+              </label>
+            </button>
+          </div>
+
+          <div className="tags-cell">
+            <span id="tags-label">Tags</span>
+            <ReactTags
+              tags={tags}
+              suggestions={tagOptions}
+              handleDelete={this.handleDelete}
+              handleAddition={this.handleAddition}
+              handleDrag={this.handleDrag}
+              delimiters={delimiters}
+              inline
+              placeholder="add tags here"
+              classNames={{
+                tags: 'tag-container',
+                tagInput: 'tag-input-container',
+                tagInputField: 'tag-input',
+                selected: 'tag-selection',
+                tag: 'tag-item',
+                remove: 'tag-remove',
+                suggestions: 'tag-menu-container',
+                activeSuggestion: 'tag-menu-item active'
+              }} />
+          </div>
+
+          <div className="save-draft-cell">
```

Comment:
Why are we calling the btns cell?

Reply:
The way I designed the grid, the last two cells are for those buttons only